src/SqlPersistence/Config/SqlDialect_Oracle.cs

```diff
@@ -0,0 +1,56 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Data;
+    using System.Data.Common;
+    using System.Text;
+
+    public abstract partial class SqlDialect
+    {
+        /// <summary>
+        /// Oracle
+        /// </summary>
+        public partial class Oracle : SqlDialect
+        {
+            internal override void FillParameter(DbParameter parameter, string paramName, object value)
+            {
+                parameter.ParameterName = paramName;
+                if (value is Guid)
+                {
+                    parameter.Value = value.ToString();
+                }
+                else if (value is Version)
+                {
+                    parameter.DbType = DbType.String;
+                    parameter.Value = value.ToString();
+                }
+                else
+                {
+                    parameter.Value = value;
+                }
+            }
+
+            internal override CommandWrapper CreateCommand(DbConnection connection)
+            {
+                var command = connection.CreateCommand();
+
+                var bindByNameProperty = command.GetType().GetProperty("BindByName");
```

Comment:
Nope, I just copied the code there. I tried to optimize it here https://github.com/Particular/NServiceBus.Persistence.Sql/pull/147